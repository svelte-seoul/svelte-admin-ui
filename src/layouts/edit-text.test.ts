import {render, screen} from '@testing-library/svelte';

import Edittext from './edit-text.svelte';
import {setUpUser} from '../../test/testUtils';

describe('render and props test', () => {
  it('should render without error', () => {
    const testingLib = render(Edittext);

    expect(testingLib.container).toMatchSnapshot();
  });

  it('has placeholder props', () => {
   const {getByRole}  = render(Edittext, {placeholder: 'test'});

   expect(getByRole('textbox')).toHaveAttribute('placeholder', 'test');
  });

  it('has type props', () => {
    const {getByAltText}  = render(Edittext, {type: 'number'});

    expect(getByAltText('alt-input')).toHaveAttribute('type', 'number');
  });

  it('has containerStyle props', () => {
    const {getByTestId}  = render(Edittext, {containerStyle: "background-color: red"});

    const container = getByTestId('container');

    expect(container).toHaveStyle('background-color: red');
  });

  it('has inputStyle props', () => {
    const {getByAltText}  = render(Edittext, {inputStyle: "color: red"});

    const input = getByAltText('alt-input');

    expect(input).toHaveStyle('color: red');
  });

  it('has errorText props', () => {
    const {getByText}  = render(Edittext, {errorText: 'error message'});

    expect(getByText('error message')).toBeTruthy();
  });
});

describe('interaction test', () => {
  it('value props should be same with value which typed in input', async () => {
    const {user} = setUpUser(Edittext);

    const input = screen.getByAltText('alt-input');
    
    await user.type(input, '1234');

    expect(input).toHaveValue('1234');
  });
});