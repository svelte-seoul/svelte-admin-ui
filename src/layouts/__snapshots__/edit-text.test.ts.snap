// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render and props test should render without error 1`] = `
<body>
  <div>
    <div
      class="edit-text svelte-xguohu"
      data-testid="container"
      style=""
    >
      <div
        class="container svelte-xguohu"
      >
        <div />
         
        <input
          alt="alt-input"
          class="svelte-xguohu"
          placeholder=""
          style=""
          type="text"
        />
         
        <div />
      </div>
       
    </div>
  </div>
</body>
`;
